### PR TITLE
Tweaks to secret domain state

### PR DESCRIPTION
**Pull request by manadart**
This change is mechanical only and is constituted by:
- Indentation of SQL/DML according to our agreed-upon standards.
- Shortened line lengths - 120 for code, and 80 for comments.
- Corrected method names in comments.
- Removal of redundant object creation for supply as SQLair type samples.
- Removal of redundant slice creation for appending to SQLair type samples.

### CODE REVIEW

File: domain/secret/state/state.go [hunk 1]
```diff
@@ -445,38 +466,37 @@ func (st State) updateSecret(
...
-     so.label) AS (&secretOwner.*)
-FROM secret_metadata sm, rev
+WITH rev AS (
+    SELECT MAX(revision) AS latest_revision
+    FROM   secret_revision
+    WHERE  secret_id = $secretID.id
+)
+SELECT (sm.secret_id,
+       version,
+	   description,
```
Comment (tlm) [inline domain/secret/state/state.go]: Fix me please :)

File: domain/secret/state/state.go [hunk 2]
```diff
@@ -1495,7 +1529,8 @@ AND    suc.unit_uuid = $secretUnitConsumer.unit_uuid
 	}
 
 	if len(dbConsumers) == 0 {
-		return nil, fmt.Errorf("secret with label %q for unit %q not found%w", label, unitName, errors.Hide(secreterrors.SecretNotFound))
+		return nil, fmt.Errorf(
+			"secret with label %q for unit %q not found%w", label, unitName, errors.Hide(secreterrors.SecretNotFound))
```
Comment (tlm) [inline domain/secret/state/state.go:1533]: Should you break the trailing `)` onto a new line?

File: domain/secret/state/state.go [hunk 3]
```diff
@@ -1504,9 +1539,12 @@ AND    suc.unit_uuid = $secretUnitConsumer.unit_uuid
...
 }
 
-// ListExternalSecretRevisions returns the secret revisions which are stored externally in
-// a secret backend, returning an error satisfying [secreterrors.SecretNotFound] if the secret does not exist.
-func (st State) ListExternalSecretRevisions(ctx context.Context, uri *coresecrets.URI, revs ...int) ([]coresecrets.ValueRef, error) {
+// ListExternalSecretRevisions returns the secret revisions which are stored
+// externally ina secret backend, returning an error satisfying
+// [secreterrors.SecretNotFound] if the secret does not exist.
+func (st State) ListExternalSecretRevisions(
+	ctx context.Context, uri *coresecrets.URI, revs ...int) ([]coresecrets.ValueRef, error,
```
Comment (tlm) [inline domain/secret/state/state.go:1546]: This formatting doesn't match the other line breaks above.

File: domain/secret/state/state.go [hunk 4]
```diff
@@ -1964,15 +2018,18 @@ WHERE  rev.secret_id = $secretInfo.secret_id`
 		return nil, 0, errors.Trace(domain.CoerceError(err))
 	}
 	if len(dbSecretConsumers) == 0 {
-		return nil, latestRevision, fmt.Errorf("secret consumer for %q and unit %q%w", uri.ID, unitName, secreterrors.SecretConsumerNotFound)
+		return nil, latestRevision, fmt.Errorf(
+			"secret consumer for %q and unit %q%w", uri.ID, unitName, secreterrors.SecretConsumerNotFound)
```
Comment (tlm) [inline domain/secret/state/state.go:2022]: This needs to be changed. I realise that it is not part of the PR but it is wrong.

```
fmt.Errorf("%w for id %q and unit %q", secreterrors.SecretConsumerNotFound, uri.ID, unitName)
```

**Review by tlm — COMMENTED** [review]
My look thus far.

File: domain/secret/state/state.go [hunk 5]
```diff
@@ -608,19 +628,21 @@ WHERE sm.secret_id = $secretID.id
...
+// revision is not the latest revision and there are no consumers for the
+// revision.
 func (st State) markObsoleteRevisions(ctx context.Context, tx *sqlair.TX, uri *coresecrets.URI) error {
 	query, err := st.Prepare(`
 WITH in_use AS (
     -- revisions that have local consumers.
     SELECT DISTINCT current_revision FROM secret_unit_consumer suc
     WHERE  suc.secret_id = $secretRef.secret_id
-UNION
+    UNION
```
Comment (wallyworld) [inline domain/secret/state/state.go:641]: I really prefer the UNION alignment as it was as it better visually separates the different sql selects

File: domain/secret/state/state.go [hunk 6]
```diff
@@ -445,38 +466,37 @@ func (st State) updateSecret(
...
-     (so.owner_kind,
-     so.owner_id,
-     so.label) AS (&secretOwner.*)
-FROM secret_metadata sm, rev
+WITH rev AS (
+    SELECT MAX(revision) AS latest_revision
+    FROM   secret_revision
+    WHERE  secret_id = $secretID.id
+)
+SELECT (sm.secret_id,
```
Comment (wallyworld) [inline domain/secret/state/state.go:474]: The original formatting with the select on its own line is visually much nicer as all the columns have consistent white space to their left
